### PR TITLE
Read/write files relative to binary instead of cwd

### PR DESCRIPTION
**Pull request by luxaritas**
Rationale: This allows for starting the server in other directories than `build`. Eg, from the root of the repository you could run `./build/MasterServer`, or from `/home/user` I could call `/opt/DarkflameServer/build/MasterServer`.

So far this has been tested by ensuring the server correctly starts from both `build` and `build/run` under Linux. Additional testing is needed, especially on other OSes. The actual code for getting the directory the binary is in is partially pulled from Boost, per the comment in `BinaryPathFinder.cpp` (there is evidently no consistent cross-platform way to do this).

### CODE REVIEW

**Comment by luxaritas** [comment]
I attempted to find instances of relative paths by searching for `./` across the codebase, but there are some instances where relative paths are not prefixed with `./` - eg, config was one situation where this happened.

**Comment by luxaritas** [comment]
Noticed the CI failures - I'll make sure those are resolved

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — COMMENTED** [review]
Looks reasonable to me - 1 question about an include

File: dCommon/BinaryPathFinder.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#ifndef __BINARYPATHFINDER__H__
+#define __BINARYPATHFINDER__H__
+
+#include <string>
```
[the remark this answers is not in the document]
Comment (luxaritas) [reply dCommon/BinaryPathFinder.h:6]: Yeah I think you're right - some stuff changed from strings to paths

**Review by EmosewaMC — APPROVED** [review]

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**